code/game/objects/items/stacks/fifty_spawner.dm

```diff
@@ -8,13 +8,17 @@
 	var/obj/item/stack/type_to_spawn = null
 
 /obj/fiftyspawner/New()
-	//spawns the 50-stack and qdels self
-	..()
-//	var/obj_path = text2path("/obj/item/stack/[material]")
-	var/obj/item/stack/M = new type_to_spawn(src.loc)
-	M.amount = M.max_amount //some stuff spawns with 60, we're still calling it fifty
-	M.update_icon() // Some stacks have different sprites depending on how full they are.
-	qdel(src)
+	spawn()
+		//spawns the 50-stack and qdels self
+		..()
+		if(istype(src.loc, /obj/structure/loot_pile)) //Spawning from a lootpile is weird, need to wait until we're out of it to do our work.
```

Comment:
This spawn() in new, and the while loop that sleeps is not so great.  However we don't use /obj/structure/loot_pile so its not urgent to fix.